lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -389,7 +389,7 @@ def tree_entry(commit, path)
 
         pathname.each_filename do |dir|
           if tmp_entry.nil?
-            tmp_entry = commit.tree[dir]
+            return nil unless (tmp_entry = commit.tree[dir])
```

Comment:
I think we also want to `return nil` if this `tmp_entry` ends up being `nil`, otherwise we can still have path `foo/bar/baz` return the entry for `/baz` if `/foo` exists, but `/foo/bar` doesn't.